### DISCUSSION TITLE
Why prisma dropped entire database because of 2 column changes in 1 table?

### DISCUSSION THREAD

**Discussion by webmonch** [prisma/prisma#21302] [Q&A]
### Question

Hey fellow developers,

in my project I basically changed 2 columns on table User, after that ran **prisma push**, and not entirely sure why - it decided to recreate all existing tables:

```
Made the column `out_from_time` on table `User` required, but there are 14 existing NULL values.
Made the column `to_from_time` on table `User` required, but there are 14 existing NULL values.


✔ To apply this change we need to reset the database, do you want to continue? 
```

I'd expect it to update only the table that was changed. Am I missing something here?

**Comment by nurul3101** [comment]
Hello @webmonch 👋 

From the error message, it looks like you made `out_from_time` and `to_from_time` columns as required, which were optional before.

There would have been existing records with `NULL` values in both of these columns that would violate these new constraints, so Prisma is suggesting to reset the database to enforce the new schema constraints, which is why it's mentioning the need to recreate all existing tables.

To avoid this, you can temporarily remove the NOT NULL constraint, update or delete the records that have NULL values in the affected columns, and then re-add the NOT NULL constraint.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by webmonch** [reply]
Thanks @nurul3101 for the explanations, but I still don't understand why delete all tables if not null was enforced on 2 columns that were not referenced anywhere else except one table.

**Reply by nurul3101** [reply]
Hmm, so your suggestion would be to reset only one table and not all tables, correct?
I think this could be an improvement which we could be helpful.
